### ISSUE TITLE
Spellcheck is omitted when a punctuation symbol ends line before the ``` in markdown

### ISSUE THREAD

**Issue by luav** [bartosz-antosik/vscode-spellright#163]
Spellcheck is omitted if a line ends with a punctuation symbol (`:;,.` and might be some others) before the code block in markdown file, for example:

<pre>
BibTeX
```bibtex
some text
```
asdf     // OK, asdf is identified as a spelling error
</pre>

But if the colon (or some other punctuation signs) is used after the `BibTeX` then ALL the following text after the code block is not spellchecked:

<pre>
BibTeX:
```bibtex
some text
```
asdf     // !!! The spell checking omits `asdf` and all the following text until the next code block
</pre>

Spellcheck version: 2.4.9 (but the same issue occurred in 3.x versions)

**Comment by bartosz-antosik** [comment]
Somehow I cannot reproduce this:

![image](https://user-images.githubusercontent.com/20492548/40232054-9e84558c-5a9d-11e8-9079-65c57f8b57da.png)

I thought, maybe CRLF/LF matters, but it does not seem to matter.

Please, you have to provide some more clue.

**Comment by luav** [comment]
Yes, it was caused by the following regexp:
```json
"spellright.ignoreRegExps": [ "/(\\b[_A-Za-z]+[_A-Z]+\\w*)/g" ],
```
which should omit the `BibTeX` term (any camelCase and snake_case terms) and leave everything else untouched...

Can you reproduce the issue with this regexp enabled?

**Comment by bartosz-antosik** [comment]
For the expression to work it should look like this (backslashes should be double quoted because of JSON storage, have a peek to README to see more examples):

`"spellright.ignoreRegExps": [ "/(\\\\b[_A-Za-z]+[_A-Z]+\\\\w*)/g" ]`

But, i am not sure if it works as you intended it:

https://regex101.com/r/jYsuqO/2

If this does nto explain please carry on/reopen the thread.

**Comment by luav** [comment]
The regexp works exactly as intended and covers all non-pure lowercase variable names as it shown by you in the playground. And the regexp is already double quoted since the original non-double quoted version is: `"/\b[_A-Za-z]+[_A-Z]+\w*/g"`.

Anyway, double double quoting (`"/(\\\\b[_A-Za-z]+[_A-Z]+\\\\w*)/g"`) as you specified causes the same issues in the markdown files as reported for the `"/\\b[_A-Za-z]+[_A-Z]+\\w*/g"`.  
So, either I missed something or there is a bug in the regexp processing by your extension, which somehow affects the following code block (probably, concatenating it with the leading punctuation sign).

**Comment by luav** [comment]
It seems I found the issue and this is a bug. You can't reproduce it because it seems you use Windows with `\r\n` EOL but when the EOL includes only one symbol then this issue appears.
Look at  these examples with double double quoted regexp and exactly the same settings:
![spellrightregexpeol_bug](https://user-images.githubusercontent.com/6162969/40263749-3db20a7c-5b17-11e8-900e-df7a6fe77d5b.png)

Then if I add an empty line so as we have 2 EOL = `\n\n` then the spelling starts working correctly except the lines synchronization loss for the second spelling hint, which was reported earlier in another issue.
![spellrightregexpeol](https://user-images.githubusercontent.com/6162969/40263769-79e42ade-5b17-11e8-82db-68b57162de70.png)

You should check EOLs processing and treat `\r`, `\r\n`, `\n` equally as a single EOL.

**Comment by bartosz-antosik** [comment]
I have tested this with LF/CRLF as I have noted in the first comment above (BTW \r does not exist any more, it was on old macOS but it is gone now and VSCode supports only the later two). But the reason for what happens is actually quite distant and is related to an expression that detects one line code insertions in markdown (a line indented by 4+ spaces). You can see that if you would put anything pass the colon (:) character in first line it would not get spelled either.

Never mind. It should be sorted out in next release.

**Comment by bartosz-antosik** [comment]
Should be resolved in version 2.4.10
